src/expr/src/vector_op/cmp.rs

```diff
@@ -118,6 +119,44 @@ where
     }
 }
 
+#[inline(always)]
+fn struct_cmp<F>(l: StructRef, r: StructRef, func: F) -> Result<bool>
```

Comment:
```suggestion
/// The comparison function for the reference types.
fn ref_item_cmp<T, F>(l: T, r: T, func: F) -> Result<bool>
where
    F: FnOnce(T, T) -> bool,
{
}
```

Please take a look if this works for both &str and StructRef.